### DISCUSSION TITLE
Load BMP failed

### DISCUSSION THREAD

**Discussion by qtbui159** [libvips/libvips#3769] [Q&A] [answered]
I have a 32bit c program use this code.
```
VipsImage* in1 = vips_image_new_from_file("a1.bmp", nullptr); //a1.bmp is 181MB
VipsImage* in2 = vips_image_new_from_file("a2.bmp", nullptr); //a2.bmp is 350MB
```

in2 is NULL, so i call vips_error_buffer.I got error message like 

magick2vips: libMagick error: MemoryAllocationFailed "a2.bmp" @ error/bmp.c/ReadBMPImage/1001
magick2vips: unable to read file "a2.bmp"

How can i fix this error?

**Comment by jcupitt** [comment]
Hi @qtbui159, can you share the file that's failing to load?

**Reply by qtbui159** [reply]
thx for your fast replying @jcupitt . I found this error cause by a1.bmp. This code can reproduce the error.
```
VipsImage* in1 = vips_image_new_from_file("a1.bmp", nullptr);
VipsImage* in2 = vips_image_new_from_file("a1.bmp", nullptr);
```
download [a1.bmp](https://drive.google.com/file/d/1F0oxEVmMBScItS3J-VWGxRRRaQ0kLkN7/view?usp=sharing)

**Reply by qtbui159** [reply]
![image](https://github.com/libvips/libvips/assets/44298862/33234efd-65de-43c3-beb2-d849860206b3)
it's all code. 
IDE: VS2022 community
System: Windows 10 x64 Home Edtion 22H2

**Reply by jcupitt** [reply]
Ah, the file is large, so imagemagick has rejected it.

You need to raise the imagemagick memory limit in `policy.xml`.

**Reply by qtbui159** [reply]
I edit policy.xml and copy whole ImageMagick-6 folder to dll path. Unfortunately it not works.
```
<policy domain="resource" name="memory" value="8GiB"/>
```

**Reply by jcupitt** [reply]
I made a windows VM and it seems to work for me without any modification:

```
jcupi@DESKTOP-HGI6HBR MINGW64 /f
$ unzip -qq Downloads/vips-dev-w64-all-8.15.0.zip

jcupi@DESKTOP-HGI6HBR MINGW64 /f
$ cd vips-dev-8.15/

jcupi@DESKTOP-HGI6HBR MINGW64 /f/vips-dev-8.15
$ ./bin/vipsheader.exe /f/Pictures/a1.bmp
F:/Pictures/a1.bmp: 11928x7984 uchar, 3 bands, srgb, magickload

jcupi@DESKTOP-HGI6HBR MINGW64 /f/vips-dev-8.15
$ ./bin/vips.exe magickload /f/Pictures/a1.bmp x.jpg

jcupi@DESKTOP-HGI6HBR MINGW64 /f/vips-dev-8.15
$ ./bin/vips.exe copy /f/Pictures/a1.bmp x.jpg

jcupi@DESKTOP-HGI6HBR MINGW64 /f/vips-dev-8.15
$
```

Which zip did you download?

**Reply by jcupitt** [reply]
Ahhhh! I just noticed you wrote:

> I have a 32bit c program use this code.

I would not use the 32-bit version unless you have a very specific need. Use the 64-bit library.

**Reply by qtbui159** [reply]
host operate system is 32bit windows. So i can only use 32-bit version. I download zip that name is vips-dev-w32-all-8.14.5.zip

**Reply by jcupitt** [reply, marked as answer]
Imagemagick needs over 2gb of RAM to load your BMP, so it's just not possible on a 32-bit machine.

You need to either use a 64-bit machine, use smaller BMP images, or use another file format. PNG would be fine, for example.

**Reply by qtbui159** [reply]
thx.got it.

**Reply by bkocev** [reply]
Hi,

is there way to load bmp images using the web distribution of libvips? I.e., without using the "*all" distribution, which depends on ImageMagic?

Thanks,
Bojan.

**Reply by jcupitt** [reply]
Sorry, no, libvips uses imagemagick to load BMP.
